CLI/klish/Makefile

```diff
@@ -3,10 +3,18 @@ SHELL = /bin/bash
 .SHELLFLAGS += -e
 
 KLISH_VERSION = 2.1.4
-PYTHONVER=3.7
 
 KLISH_SRC = $(SONIC_CLI_ROOT)/klish-$(KLISH_VERSION)
 
+# Python changed how to link against Python for embedding purposes in 3.8
+# which means that in order to support both Debian 10 and Debian >=11
+# we need to try two alternatives.
+CFLAGS_PYTHON := $(shell pkg-config --silence-errors --cflags python3-embed || pkg-config --cflags python3)
+LDFLAGS_PYTHON := $(shell pkg-config --silence-errors --libs python3-embed || pkg-config --libs python3)
+# The installed libcurl's -dev package is not gnutls
+# so we have to explicitly link to the correct .so file
+LDFLAGS_CURL := -l:libcurl-gnutls.so.4
```

Comment:
Why do we need the gnutls version of libcurl here?

Reply:
Never mind, I just read the PR description.

Does linking fail if you just say `-lcurl` instead of `-l:libcurl-gnutls.so.4`?

Reply:
> Never mind, I just read the PR description.
> 
> Does linking fail if you just say `-lcurl` instead of `-l:libcurl-gnutls.so.4`?

It doesn't fall, it links to libcurl openssl (which falls the build in a later stage as there are hard coded tests that check for gnutls).

In our own fork we use openssl instead of gnutls as we couldn't see a reason why you explicitly link against gnutls, and the rest of SONiC uses openssl. However, just because it works for us doesn't mean it works for everyone so I haven't pushed that change here - surely somebody had a reason to explicitly link against gnutls here at some point, but I don't know why.

Reply:
I'm in favor of changing it to use openssl instead of gnutls. I don't see anything specific in the PR that added that dependency that would require gnutls.

@kwangsuk could you chime in on whether there was a specific reason to use the gnutls backend of libcurl, or was it for convenience?

Reply:
In the past, a built-in option "clish_restcl" was introduced along with the embedded python option "clish_pyobj". 
FYI - https://github.com/sonic-net/SONiC/blob/master/doc/mgmt/Management%20Framework.md#32215-actione

During integration, the gnutls was linked to build klish binary with rest_cl plugin in our initial code.
Yes, I agree that it's better to link openssl for consistent use of openssl/curl in SONiC.

@bluecmd 
As the Issue #130 was already opened for this, can you please update accordingly to use openssl?

Reply:
@kwangsuk Since it's a larger scope of the PR, I created https://github.com/sonic-net/sonic-mgmt-framework/pull/133 as an alternative. I do not want to modify this PR to preserve the discussion as in my experience GitHub is pretty bad at presenting older commits that have been force pushed.

Reply:
@bluecmd Okay, understood. Let's take #133 in context.